### ISSUE TITLE
progressbar does not honor bar_format when there is no total

### ISSUE THREAD

**Issue by BoPeng** [tqdm/tqdm#343]
I would like to use `tqdm` to generate a simple status updating line, basically creating a progress bar with only `{desc}` at line 0,

```
prog = tqdm(desc=msg, position=0, bar_format='{desc}')
```

and update it from time to time with

```
prog.set_description(updated_desc)
prog.update(1)
```

However prog would print out time etc, despite of the parameter `bar_format='{desc}'`. Digging through the code I found that [parameter `bar_format` is not used when there is no `total`](https://github.com/tqdm/tqdm/blob/master/tqdm/_tqdm.py#L368)  so I cannot control the output in this case.

I can get around of this problem by setting a fake `total`, e.g.,

```
prog = tqdm(desc=msg, position=0, total=10000, bar_format='{desc}')
```

but this might be something you would like to fix.

**Comment by BoPeng** [comment]
BTW, the fact that tqdm [automatically adds `':'` after `desc`](https://github.com/tqdm/tqdm/blob/master/tqdm/_tqdm.py#L706) made this usage slightly awkward because it would show `done.:` with `set_description('done.')`.

**Comment by casperdcl** [comment]
Thanks @BoPeng. Your workaround of having a fake large `total` is pretty good. We really should be supporting a custom `bar_format` when `total` is not available.

**Comment by jnatten** [comment]
A workaround for the added `: ` is to just use `\b` and print whitespace over it.
So forexample a bar with just a description would look like: 
`tqdm(position=2, total=100, bar_format="{desc}\b\b ")`

**Comment by casperdcl** [comment]
no, just use `set_description_str()` instead of `set_description()`

**Comment by jnatten** [comment]
Thats smarter :)
